contrib/platform/src/com/sun/jna/platform/win32/WTypes.java

```diff
@@ -160,15 +164,29 @@ public BSTRByReference(BSTR value) {
         }
 
         public void setValue(BSTR value) {
+            // Save a local reference to prevent losing value's memory
+            this.bstr = value;
             this.getPointer().setPointer(0, value.getPointer());
         }
 
         public BSTR getValue() {
-            return new BSTR(getPointer().getPointer(0));
+            // If memory address hasn't changed, return the stored BSTR
+            if (bstr != null) {
+                if (getPointer().getPointer(0).equals(bstr.getPointer())) {
+                    return bstr;
+                }
+                // The stored memory has been changed, memory is tracked elsewhere.
+                // Clean up unneeded reference.
+                bstr = null;
+            }
+            // Null check the pointer
+            Pointer p = getPointer().getPointer(0);
+            return p != null ? new BSTR(p) : null;
```

Comment:
Either this null check is unnessary, or there is a potential problem in line 175.

Reply:
If `bstr` is null we never check line 175 and jump straight here.  I believe this is the normal case when `new BSTRByReference()` is instantiated before any value is set (e.g., before `setValue()` is called)